frontend/src/modules/machinery/components/Machinery.js

```diff
@@ -90,24 +106,48 @@ export default class Machinery extends React.Component<Props> {
                                 id='machinery-search'
                                 type='search'
                                 autoComplete='off'
-                                placeholder='Search in Machinery'
+                                placeholder='Concordance Search'
                                 ref={this.searchInput}
                             />
                         </Localized>
                     </form>
                 </div>
-                <ul>
-                    {machinery.translations.map((translation, index) => {
-                        return (
-                            <Translation
-                                index={index}
-                                sourceString={machinery.sourceString}
-                                translation={translation}
-                                key={index}
-                            />
-                        );
-                    })}
-                </ul>
+                <div className='list-wrapper'>
+                    <ul>
+                        {machinery.translations.map((translation, index) => {
+                            return (
+                                <Translation
+                                    index={index}
+                                    sourceString={machinery.sourceString}
+                                    translation={translation}
+                                    key={index}
+                                />
+                            );
+                        })}
+                    </ul>
+                    <ul>
+                        {machinery.searchResults.map((result, index) => {
+                            return (
+                                <Translation
+                                    index={index}
```

Comment:
I haven't tested this myself but I think this will create a problem with the `ctrl+shift+up/down` keyboard shortcuts, which rely on indexes to select a prev/next result from helpers. The idea is that indexes should be consecutive, and they shouldn't overlap. To fix this, you should be able to offset this index with the amount of machinery translations.

Additionally, the value of `key` can also create other issues ([see how React uses the `key` prop](https://reactjs.org/docs/lists-and-keys.html#keys)), especially because (and correct me if I'm wrong) search results are sorted as they arrive. Do we have a uniquely-identifying value for every search result that we could use as a `key`?

Reply:
I didn't love using the index for the key, but sadly the only possible 'unique' value will be the translation which can be very long in some cases, and I'm also not sure if it will always be unique as the items are only combined into a single value if both the original string and the translation are the same. I suppose there could be times when the original might be different but the translation would be the same.

With that in mind I will offset the index value to avoid conflict with the shortcut.

Reply:
After taking a closer look here, the use of `index` was how this was originally set up when only mapping the `translations`. So, I tested the shortcut you mentioned, and it still works as expected.

Reply:
I tested this on my end and I'm afraid it doesn't work as expected 😬. Please check the following screencast, where all taken shortcut actions are `ctrl+shift+down`.

https://user-images.githubusercontent.com/16768/107953295-3d0fed00-6f9b-11eb-8975-53bc0cc02418.mov

You'll notice 2 bugs:
* Performing concordance search leaves the element at previously-selected index highlighted. The easy fix would be to reset the selected index upon search (`editor.actions.resetHelperElementIndex()`).
* Cycling through search results doesn't respect visual ordering. I'm almost certainly sure this is related to the original `index` problem I was mentioning earlier here.

Reply:
Ah, yes...... I fully see the problem now.  Thanks for your patience 😊